resources/js/features/achievements/components/UnlockableAchievementAvatar/UnlockableAchievementAvatar.tsx

```diff
@@ -0,0 +1,77 @@
+import type { FC } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { AchievementAvatar } from '@/common/components/AchievementAvatar';
+import { GameAvatar } from '@/common/components/GameAvatar';
+import type { AvatarSize } from '@/common/models';
+import { UnlockedAtLabel } from '@/features/achievements/components/UnlockedAtLabel';
+
+interface UnlockableAchievementAvatarProps {
+  achievement: App.Platform.Data.Achievement;
+  showGame?: boolean;
+  imageSize?: AvatarSize;
+}
+
+export const UnlockableAchievementAvatar: FC<UnlockableAchievementAvatarProps> = ({
+  achievement,
+  showGame = false,
+  imageSize = 48,
+}) => {
+  const { t } = useTranslation();
+
+  return (
+    <div className="mb-2 flex items-center gap-2">
+      {achievement.unlockedHardcoreAt ? (
+        <AchievementAvatar
+          {...achievement}
+          showHardcoreUnlockBorder={true}
+          // TODO: showPointsInTitle={true}
+          showLabel={false}
+          size={imageSize}
+        />
+      ) : achievement.unlockedAt ? (
+        <AchievementAvatar
+          {...achievement}
+          showHardcoreUnlockBorder={false}
+          // TODO: showPointsInTitle={true}
+          showLabel={false}
+          size={imageSize}
+        />
+      ) : (
+        <AchievementAvatar
+          {...achievement}
+          showHardcoreUnlockBorder={false}
+          badgeUnlockedUrl={achievement.badgeLockedUrl}
+          // TODO: showPointsInTitle={true}
+          showLabel={false}
+          size={imageSize}
+        />
+      )}
```

Comment:
Here's a good way to trim all this down:
```tsx
const badgeUrl =
  !achievement.unlockedAt && !achievement.unlockedHardcoreAt
    ? achievement.badgeLockedUrl
    : achievement.badgeUnlockedUrl;

return (
  <div className="mb-2 flex items-center gap-2">
    <AchievementAvatar
      {...achievement}
      showHardcoreUnlockBorder={!!achievement.unlockedHardcoreAt}
      badgeUnlockedUrl={badgeUrl}
      // TODO: showPointsInTitle={true}
      showLabel={false}
      size={imageSize}
    />
```

We're overloading `badgeUnlockedUrl` in `<AchievementAvatar />`, but we can figure that out later.

Reply:
Done.

I tried doing that in the {} without much luck. Didn't think to do it outside the return.